msal/application.py

```diff
@@ -1417,12 +1417,18 @@ def acquire_token_by_username_password(
             user_realm_result = self.authority.user_realm_discovery(
                 username, correlation_id=headers[msal.telemetry.CLIENT_REQUEST_ID])
             if user_realm_result.get("account_type") == "Federated":
-                response = _clean_up(self._acquire_token_by_username_password_federated(
-                    user_realm_result, username, password, scopes=scopes,
-                    data=data,
-                    headers=headers, **kwargs))
-                telemetry_context.update_telemetry(response)
-                return response
+                try:
+                    response = _clean_up(self._acquire_token_by_username_password_federated(
+                        user_realm_result, username, password, scopes=scopes,
+                        data=data,
+                        headers=headers, **kwargs))
+                except (ValueError, RuntimeError):
```

Comment:
https://github.com/Azure/azure-cli/issues/21068#issue-1109567002 is an `AssertionError`. Shouldn't we catch it as well?

Reply:
Nah.

You may know that the `AssertionError` is not reliable in Python. Previously I used it [here](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/456/files#diff-dd5486a6b885e238cc315b1ad8820acb0a7ccbde95fbcaee8a091825e58115c9L47) anyway, and that's why I also made it clear with an explicit inline comment "a loose check".

This time, since we are trying to use that exception to provide some useful info, I already switched it to `ValueError` in this PR, too.

Reply:
Then perhaps explicitly use Exception Chaining (https://www.python.org/dev/peps/pep-3134/)?

```
raise RuntimeError ... from ex
```

Reply:
Let's just settle with the implicit chaining for now. :-)

So, overall I guess I can deduce that you think the exception messages are fine. I'm going to merge this in tomorrow, and then start the release process.